crates/bevy_asset/src/assets.rs

```diff
@@ -465,19 +455,27 @@ impl<A: Asset> Assets<A> {
         let mut infos = asset_server.data.infos.write();
         let mut not_ready = Vec::new();
         while let Ok(drop_event) = assets.handle_provider.drop_receiver.try_recv() {
-            let id = drop_event.id;
-            if !assets.contains(id.typed()) {
+            let untyped_id = drop_event.id.untyped(TypeId::of::<A>());
+            let id = drop_event.id.typed();
+
+            // TODO: How to differentiate between the asset having been unloaded,
```

Comment:
@cart Any idea how to fix this?

Current design for this PR is to get rid of Assets::remove(). Assets::remove_untracked() is now Assets::unload(). 

Now the removal event is only sent when the last strong handle is dropped, regardless of whether the asset was already unloaded or not.

Reply:
We could probably query the AssetServer somehow to check if the asset is loading or not. I don't know how to properly do this though.